src/beamlime/applications/handlers.py

```diff
@@ -23,17 +22,22 @@ class DataReductionHandler(HandlerInterface):
     """Data reduction handler to process the raw data."""
 
     def __init__(self, workflow: StatelessWorkflow) -> None:
+        from ._nexus_helpers import NexusContainer
+
+        self.nexus_container: NexusContainer
         self.workflow = workflow
         super().__init__()
 
-    def format_received(self, data: Any) -> str:
-        # TODO remove ties to specific type of data
-        return f"{len(data)} pieces of {Events.__name__}"
+    def update_nexus_template(self, message: RunStart) -> None:
+        self.nexus_container = message.content
+        self.info("Received nexus template, ready to process the data.")
 
-    def process_message(self, message: MessageProtocol) -> MessageProtocol:
-        content = message.content
-        self.info("Received, %s", self.format_received(content))
-        return WorkflowResultUpdate(content=self.workflow(content))
+    def process_message(self, message: DetectorDataReceived) -> WorkflowResultUpdate:
+        self.info("Received data from %s.", message.content["source_name"])
+        self.nexus_container.insert_ev44(message.content)
+        return WorkflowResultUpdate(
+            content=self.workflow(self.nexus_container.nexus_dict)
+        )
```

Comment:
This makes the stateless workflow handler stateful. This seems to undo what we have worked towards in the past couple of weeks.

- The stateless workflow must be called on data that contains something akin to a time-slice of an NeXus file. We probably do not want to do this on something where only a single ev44 update is included.
- Please read the description in #134 (which this PR is supposed to fix). The message merging should *not* be part of the workflow. It is done in, e.g., the (Fake)Listener, or possibly a step in between.

Aside from the above, can you minimize package-wide changes that are unrelated to the PR? It makes reviewing really hard.